### PR TITLE
PR to finish SiteQC pipeline: adds triodata, conditional rule execution, upgrades aggrAnnot

### PR DESCRIPTION
**Pull request by Vlad-Dembrovskyi**
### This PR does the following:
- [x] Adds `triodata_define` process and R script ([#9987](https://app.zenhub.com/workspaces/lifebit-platform-5aa6276b1224bb03d13f0e82/issues/lifebit-ai/lifebit-platform/9987))
- [x] Makes `mend_err_p2` and `mend_dist` processes executed conditionally on input files ([#9989](https://app.zenhub.com/workspaces/lifebit-platform-5aa6276b1224bb03d13f0e82/issues/lifebit-ai/lifebit-platform/9989))
- [x] Makes `triodata_define` processes executed conditionally on input files ([#9991](https://app.zenhub.com/workspaces/lifebit-platform-5aa6276b1224bb03d13f0e82/issues/lifebit-ai/lifebit-platform/9991))
- [x] Updates `aggregate_annotation` process and R script according to changes from Daniel from 15th of september ([#10159](https://app.zenhub.com/workspaces/lifebit-platform-5aa6276b1224bb03d13f0e82/issues/lifebit-ai/lifebit-platform/10159))

### To test
First, pull the proper version of pipeline:
```
git clone https://github.com/lifebit-ai/siteqc.git
cd siteqc
git checkout vlad-dev2
```
**Test 1:** run the full pipeline with all of the processes:
(will take ~4 minutes)
```
nextflow run main.nf --input s3://lifebit-featured-datasets/projects/gel/siteqc/input.csv
```
**Test 2:** run the pipeline with minimum processes possible by providing needed files for two conditional executions:
```
nextflow run main.nf \
    --input s3://lifebit-featured-datasets/projects/gel/siteqc/input.csv \
    --triodata_keep_pheno s3://lifebit-featured-datasets/projects/gel/siteqc/keep.txt \
    --triodata_fam s3://lifebit-featured-datasets/projects/gel/siteqc/example.fam \
    --mend_err_p3_keep_fam s3://lifebit-featured-datasets/projects/gel/siteqc/MendelFamilies_4SD.fam \
    -resume #optional, to save time by getting the cached big 1kg big file.
```
**Test 3 and 4** (if needed) - only files for one condition at a time:
```
nextflow run main.nf \
    --input s3://lifebit-featured-datasets/projects/gel/siteqc/input.csv \
    --triodata_keep_pheno s3://lifebit-featured-datasets/projects/gel/siteqc/keep.txt \
    --triodata_fam s3://lifebit-featured-datasets/projects/gel/siteqc/example.fam \
    -resume #optional
```
```
nextflow run main.nf \
    --input s3://lifebit-featured-datasets/projects/gel/siteqc/input.csv \
    --mend_err_p3_keep_fam s3://lifebit-featured-datasets/projects/gel/siteqc/MendelFamilies_4SD.fam \
    -resume #optional
```
### My local test results (all processes):
image

### My test in GEL production environment:
image
image

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -226,6 +215,71 @@ log.info "-\033[2m--------------------------------------------------\033[0m-"
 //     """
 // }
 
+
+// CONDITION: if triodata_fam or triodata_keep_pheno files are not provided -
+// run triodata_define process to generate them.
+
+if (params.triodata_keep_pheno == 's3://lifebit-featured-datasets/projects/gel/siteqc/nodata' || params.triodata_fam == 's3://lifebit-featured-datasets/projects/gel/siteqc/nodata') {
```
Comment (cgpu) [inline main.nf:222]: To avoid hardcoding in the main.nf the exact file path you could do:

```diff
- if (params.triodata_keep_pheno == 's3://lifebit-featured-datasets/projects/gel/siteqc/nodata' || params.triodata_fam == 's3://lifebit-featured-datasets/projects/gel/siteqc/nodata') {
+ if (params.triodata_keep_pheno == params.empty_file || params.triodata_fam ==  params.empty_file) {
```

Comment (Vlad-Dembrovskyi) [reply main.nf]: Fixed.

**Review by cgpu — APPROVED** [review]
Great job @Vlad-Dembrovskyi this lgtm!
Thanks for adding the docs section.

Proceed to squashing 🎃 !

File: docs/usage.md [hunk 2]
```diff
@@ -39,6 +39,165 @@ First, go to the [nf-core/siteqc releases page](https://github.com/nf-core/siteq
 
 This version number will be logged in reports when you run the pipeline, so that you'll know what you used when you look back in the future.
 
+## Required SiteQC pipeline arguments
```
Comment (cgpu) [inline docs/usage.md:42]: @aldembert can you can docs as well for the `lifebit-ai/relate` subworkflow as they are added here?

Comment (aldembert) [reply docs/usage.md:42]: Sure, Im on it @cgpu

**Pull request merged**